bin/get-video-s3-acls.py

```diff
@@ -0,0 +1,324 @@
+    "--cached",
+    default=False,
+    required=False,
+    action="store_true",
+    help="Use keys generated on a previous non-cached run (default: %(default)s) "
+    "(Don't mix PRODUCTION and STAGING!)",
+)
+parser.add_argument(
+    "--pgclient",
+    default="/usr/bin/psql",
+    required=False,
+    help=f"Postgres client path (default: %(default)s)",
+)
+parser.add_argument(
+    "--awsclient",
+    default="/usr/local/bin/aws",
+    required=False,
+    help=f"AWS client path (default: %(default)s)",
+)
+args = parser.parse_args()
+
+# Globals
+AWSCLIENT = args.awsclient
+PGCLIENT = args.pgclient
+DATABASE_URL = os.getenv("DATABASE_URL", "")
+NEW_ENV = os.environ.copy()
+CSV_DELIMITER = ","
+AWS_S3_BUCKET = f"nzsl-signbank-media-{args.mode}"
+TMPDIR = "/tmp/nzsl"
+try:
+    os.makedirs(TMPDIR, exist_ok=True)
+except OSError as err:
+    print(f"Error creating temporary directory: {TMPDIR} {err}", file=sys.stderr)
+    exit()
+NZSL_POSTGRES_RAW_KEYS_FILE = f"{TMPDIR}/nzsl_postgres_raw_keys.txt"
+S3_BUCKET_RAW_KEYS_FILE = f"{TMPDIR}/s3_bucket_raw_keys.txt"
+ALL_KEYS_FILE = f"{TMPDIR}/all_keys.csv"
+
+# Vars
+nzsl_raw_keys_dict = {}
+s3_bucket_raw_keys_list = []
+all_keys_dict = {}
+
+
+# Truncate files, creating them if necessary
+def init_files(
+    files_list=(NZSL_POSTGRES_RAW_KEYS_FILE, S3_BUCKET_RAW_KEYS_FILE, ALL_KEYS_FILE)
+):
+    for p in files_list:
+        f = open(p, "a")
+        f.truncate()
+        f.close()
+
+
+# Pull all info from existing file
+def get_keys_from_cache_file(cache_file=ALL_KEYS_FILE):
+    nkeys_present = 0
+    nkeys_absent = 0
+    this_all_keys_dict = {}
+    try:
+        with open(cache_file, "r") as f_obj:
+            for line in f_obj.readlines():
+                (
+                    video_key,
+                    is_present_str,
+                    db_id_str,
+                    gloss_id_str,
+                    is_public_str,
+                ) = line.strip().split(CSV_DELIMITER)
+
+                is_present = is_present_str.strip().lower() == "true"
+                if is_present:
+                    nkeys_present += 1
+                    db_id = int(db_id_str)
+                    # Some don't have gloss_id's
+                    try:
+                        gloss_id = int(gloss_id_str)
+                    except ValueError:
+                        gloss_id = None
+                    is_public = is_public_str.strip().lower() == "true"
+                else:
+                    nkeys_absent += 1
+                    db_id = None
+                    gloss_id = None
+                    is_public = None
+
+                this_all_keys_dict[video_key] = [is_present, db_id, gloss_id, is_public]
+
+        print(f"PRESENT: {nkeys_present} keys", file=sys.stderr)
+        print(f"ABSENT:  {nkeys_absent} keys", file=sys.stderr)
+
+        return this_all_keys_dict
+
+    except FileNotFoundError:
+        print(f"File not found: {cache_file}", file=sys.stderr)
+        exit()
```

Comment:
suggestion: `open` should already raise an error if the file doesn't exist, which includes the path to the file itself, and Python prints unhandled errors to `stderr`

so I think we should just drop the `try/except` here as it'll let us greatly dedent this method

Reply:
Done!